src/components/hub/elements/ConversationItem.js

```diff
@@ -22,8 +24,17 @@ export default function ConversationItem({
     if (type === "g") {
       return <Group />;
     }
-    return chatName ? chatName.slice(0, 2).toUpperCase() : <UnknownPhoto />;
-  }, [type, chatName]);
+
+    return (
+      <UserAvatar
+        avatarUrl={chatAvatarObject.avatar_url}
+        avatarBlurHash={chatAvatarObject.file_blur_hash}
+        defaultIcon={
+          chatName ? chatName.slice(0, 2).toUpperCase() : <UnknownPhoto />
+        }
+      />
+    );
+  }, [type, chatName, chatAvatarObject]);
```

Comment:
I assume this `useMemo` does not work the way you think it should.

There is a `chatAvatarObject ` as a dependency.

But, in `src/components/hub/elements/ConversationItemList.js` component you construct it as a new object every time

```
      chatAvatarObject={{
            ...chatParticipant.avatar_object,
            avatar_url: chatParticipant.avatar_url,
          }}
```

so the `chatAvatarObject` dependency will have a new reference all the time which leads to `usememo` recalculation

Reply:
As a possible solution, instead of passing an object, we can separately pass `avatar_url ` and `file_blur_hash `  in props

Reply:
I suggest to double check the `ParticipantInChat` as well